### PR TITLE
Add F3RM Robot Code

### PR DESCRIPTION
**Pull request by williamshen-nz**
- Update datasets, added NeRF to world transformations and Panda demo datasets
- Update READMEs to include instructions for installing robot code
- Cache the features in the FeatureDataManager
- Language-Guided Optimization Code with visualization support
- Updated `pyproject.toml` with dependencies

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -46,13 +54,15 @@ f3rm = "f3rm.f3rm_config:f3rm_method"
...
+f3rm-optimize = "f3rm_robot.optimize:entrypoint"
 
 [tool.setuptools.packages]
-find = { include = ["f3rm", "f3rm.*"] }
+find = { include = ["f3rm", "f3rm.*", "f3rm_robot", "f3rm_robot.*"] }
 
 [tool.setuptools.package-data]
 "f3rm.scripts.images" = ["*.png"]  # images for demo CLIP and DINO features
 "f3rm.features.clip" = ["bpe_simple_vocab_16e6.txt.gz", "README.md"]  # CLIP assets
+"f3rm_robot" = ["README.md"]
```
Comment (williamshen-nz) [inline pyproject.toml:64]: Make sure other things in assets are packaged properly

File: pyproject.toml [hunk 2]
```diff
@@ -33,6 +33,14 @@ dev = [
     "black",
     "isort",
 ]
+robot = [
+    "open3d",
+    "params-proto",
+    "PyMCubes",
+    "python-slugify",
+    # Loosen version requirements when viser is more stable
+    "viser==0.1.7"
```
Comment (williamshen-nz) [inline pyproject.toml]: Conflicts with nerfstudio viser version

File: f3rm_robot/field_adapter.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+from typing import Dict
+
+import torch
+import torch.nn as nn
+from jaxtyping import Float
```
Comment (geyang) [inline f3rm_robot/field_adapter.py:5]: uhm, why are we using the jaxtyping Float?

File: f3rm_robot/initial_proposals.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    from mcubes import marching_cubes
+    from trimesh import Trimesh
+
+    assert alpha.dim() == 3, f"alpha should be 3D, not {alpha.shape}"
+
+    # We use mcubes as it's sufficiently fast (<0.02s). pytorch3d and kaolin are painful to deal with.
+    verts, faces = marching_cubes(alpha.cpu().numpy(), iso_value)
+
+    # Create a Trimesh, then scale and translate back to the original world coordinates
+    mesh = Trimesh(verts, faces)
```
Comment (geyang) [inline f3rm_robot/initial_proposals.py:79]: Trimesh sometimes has issue reading larger pointclounds. The way it parses the ply files tend to fail when trying to detect the chartset.

**Review by geyang — APPROVED** [review]
Looks good to me.

File: f3rm_robot/field_adapter.py (hunk 3 above)
Comment (williamshen-nz) [reply f3rm_robot/field_adapter.py:5]: That's what nerfstudio latest uses. Seems like it is better alternative to torchtyping

File: f3rm_robot/initial_proposals.py (hunk 4 above)
Comment (williamshen-nz) [reply f3rm_robot/initial_proposals.py:79]: Right that's good to know. We're initializing meshes directly from their numpy arrays here so it should be ok.

**Pull request merged**